analyzer/block/block.go

```diff
@@ -156,6 +165,72 @@ func (b *blockBasedAnalyzer) fetchBatchForProcessing(ctx context.Context, from u
 	return heights, nil
 }
 
+// Returns info about already-processed blocks in the analyzer's configured range:
+// - Whether the already-processed blocks form a contiguous range starting at "to".
+// - Whether any of the blocks were processed by fast-sync.
+// - The height of the highest already-processed block.
+func (b *blockBasedAnalyzer) processedSubrangeInfo(ctx context.Context) (bool, int64, error) {
+	var isContiguous bool
+	var maxProcessedHeight int64
+	if err := b.target.QueryRow(
+		ctx,
+		queries.ProcessedSubrangeInfo,
+		b.analyzerName,
+		b.blockRange.From,
+		b.blockRange.To,
+	).Scan(&isContiguous, &maxProcessedHeight); err != nil {
+		return false, 0, err
+	}
+	return isContiguous, maxProcessedHeight, nil
+}
+
+// Returns true if the block immediately preceding `height` has been processed in slow-sync mode.
+func (b *blockBasedAnalyzer) isBlockProcessedBySlowSync(ctx context.Context, height int64) (bool, error) {
+	var isProcessed bool
+	if err := b.target.QueryRow(
+		ctx,
+		queries.IsBlockProcessedBySlowSync,
+		b.analyzerName,
+		height,
+	).Scan(&isProcessed); err != nil {
+		return false, err
+	}
+	return isProcessed, nil
+}
+
+// Validates assumptions/prerequisites for starting a slow sync analyzer:
+//   - No blocks in the configured [from, to] range have been processed, except possibly a contiguous
+//     subrange [from, X] for some X.
+//   - If the most recently processed block was not processed by slow-sync (i.e. by fast sync, or not
+//     at all), triggers a finalization of the fast-sync process.
+func (b *blockBasedAnalyzer) ensureSlowSyncPrerequisites(ctx context.Context) (ok bool) {
```

Comment:
```suggestion
func (b *blockBasedAnalyzer) ensureSlowSyncPrerequisites(ctx context.Context) bool {
```

since the method doesn't utilized the named return.